django/template/loaders/cached.py

```diff
@@ -39,7 +51,7 @@ def get_template(self, template_name, template_dirs=None, skip=None):
                 template_name, template_dirs, skip,
             )
         except TemplateDoesNotExist as e:
-            self.get_template_cache[key] = e
+            self.get_template_cache[key] = copy_exception(e) if self.engine.debug else TemplateDoesNotExist
```

Comment:
As per my comment above I think I'd like to just move this into its own function and write a substantial docstring. The reasons for caching what I'm caching are much easier to explain in five lines than two, and I don't want to clutter this function with block comments.